openapi-generator/templates/java/libraries/okhttp-gson/ApiClient.mustache

```diff
@@ -654,6 +654,26 @@ public class ApiClient {
         }
     }
 
+    public List<Pair> mappedParameterToPairs(String name, Object parameter){
+        List<Pair> params = new ArrayList<Pair>();
+
+        if(parameter instanceof Map){
```

Comment:
there's additional problem that we'll have to address at some point: these objects can be nested. so, it should be possible to generate a parameter such as `formatOptions[customMetadataColumns][order]=B` and so on :smile:  Perhaps we could address that in a separate ticket

Reply:
I guess in case if a value against format_option is an array, it should already be [handled](https://github.com/phrase/openapi/blob/master/openapi-generator/templates/java/libraries/okhttp-gson/ApiClient.mustache#L643)? If not I will check and see if that can be addressed as well.

Reply:
seems it covers the basic case, but I guess arrays can also be nested :grimacing: never mind, we can cross that bridge when we get there

Reply:
😱

Reply:
Here's something that we can use as a reference implementation: https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/to_query.rb but I suggest leaving that for another ticket.

Reply:
Was already looking into nested arguments, that part is addressed with the latest commit. Regarding values being arrays/nested array, can see if it can be handled separately.

Reply:
@theSoenke by same issue do you mean, it will require same patch like this?? For map based query params?

Reply:
@hahmed-dev `format_options` should already work. But for more deeply nested params it will need a similar fix I think